From 055d7b88b815106a2281320668f218c012eda1ad Mon Sep 17 00:00:00 2001
From: Ralf Gommers <ralf.gommers@gmail.com>
Date: Sun, 3 Jul 2022 13:34:13 +0200
Subject: [PATCH] BUG: remove `-Dstrip=true` default, it doesn't work on macOS

See https://github.com/scipy/scipy/issues/16446 for details
on how it fails on macOS and also for proof that it is not
needed because it does not change the wheel size on Linux.

This was originally discussed in gh-27. It looks like I was
wrong there about the need for stripping. That info was based
on my experience with `numpy.distutils` builds; there it is
needed because of the hardcoded debug flags that cannot be
switched with a flag like `-Ddebug=false` like we use in
`meson-python`.
---
 mesonpy/__init__.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mesonpy/__init__.py b/mesonpy/__init__.py
index 6594812..0896e34 100644
--- a/mesonpy/__init__.py
+++ b/mesonpy/__init__.py
@@ -487,7 +487,6 @@ def _configure(self, reconfigure: bool = False) -> None:
                 f'--native-file={os.fspath(self._meson_native_file)}',
                 # TODO: Allow configuring these arguments
                 '-Ddebug=false',
-                '-Dstrip=true',
                 '-Doptimization=2',
                 *setup_args,
             )
